projects/plugins/jetpack/_inc/client/sharing/publicize.jsx

```diff
@@ -24,8 +25,24 @@ export const Publicize = withModuleSettingsFormHelpers(
 				isOfflineMode = this.props.isOfflineMode,
 				siteRawUrl = this.props.siteRawUrl,
 				isActive = this.props.getOptionValue( 'publicize' ),
+				hasSocialBasicFeatures = this.props.hasSocialBasicFeatures,
+				hasSocialAdvancedFeatures = this.props.hasSocialAdvancedFeatures,
+				isAtomicSite = this.props.isAtomicSite,
+				activeFeatures = this.props.activeFeatures,
 				userCanManageModules = this.props.userCanManageModules;
 
+			const showUpgradeLink =
+				! isAtomicSite &&
+				activeFeatures &&
+				activeFeatures.length > 0 &&
+				isActive &&
+				! hasSocialAdvancedFeatures;
+
+			const jetpackSocialText = __(
```

Comment:
Since we only use that string once, maybe we do not need to extract it into its own constant; we could display it in place right away.